tasks/src/test/scala/dagr/tasks/ScatterGatherTests.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.tasks
+
+import java.nio.file.{Files, Path}
+
+import dagr.commons.io.Io
+import dagr.commons.util.{LazyLogging, LogLevel, Logger, UnitSpec}
+import dagr.core.execsystem.{SystemResources, TaskManager}
+import dagr.core.tasksystem.{Pipeline, SimpleInJvmTask}
+import dagr.tasks.ScatterGather.{Partitioner, Scatter}
+import org.scalatest.BeforeAndAfterAll
+
+/**
+  * Tests for the scatter gather task framework.
+  *
+  * Our test involves a "complicated" little scatter/gather that does the following:
+  *   - Scatter a text file by lines
+  *   - Count the words on each line
+  *   - [Gather] Square the counts
+  *   - Sum the counts
+  *   - [Gather] Sum the squares
```

Comment:
Probably need a test or two that does:
- `Scatter, Map, Gather, Scatter, Map, Gather`
- `Scatter, Map, Gather.  Map on the original Scatter, and another Gather` (is this supported)?
You could log an issue and I'd be happy to take a shot at adding those in.